nsv13/code/modules/power/rbmk.dm

```diff
@@ -0,0 +1,840 @@
+			switch(power)
+				if(20 to 39)
+					grilled_item.name = "grilled [initial(grilled_item.name)]"
+					grilled_item.desc = "[initial(I.desc)] It's been grilled over a nuclear reactor."
+					if(!grilled_item.foodtype & FRIED)
+						grilled_item.foodtype |= FRIED
+				if(40 to 70)
+					grilled_item.name = "heavily grilled [initial(grilled_item.name)]"
+					grilled_item.desc = "[I.desc] It's been heavily grilled through the magic of nuclear fission."
+					if(!grilled_item.foodtype & FRIED)
+						grilled_item.foodtype |= FRIED
+				if(70 to 95)
+					grilled_item.name = "Three-Mile Nuclear-Grilled [initial(grilled_item.name)]"
+					grilled_item.desc = "A [initial(grilled_item.name)]. It's been put on top of a nuclear reactor running at extreme power by some badass engineer."
+					if(!grilled_item.foodtype & FRIED)
+						grilled_item.foodtype |= FRIED
+				if(95 to INFINITY)
+					grilled_item.name = "Ultimate Meltdown Grilled [initial(grilled_item.name)]"
+					grilled_item.desc = "A [initial(grilled_item.name)]. A grill this perfect is a rare technique only known by a few engineers who know how to perform a 'controlled' meltdown whilst also having the time to throw food on a reactor. I'll bet it tastes amazing."
+					if(!grilled_item.foodtype & FRIED)
+						grilled_item.foodtype |= FRIED
+
+//Method to handle sound effects, reactor warnings, all that jazz.
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/proc/handle_alerts()
+	var/alert = FALSE //If we have an alert condition, we'd best let people know.
+	if(K <= 0 && temperature <= 0)
+		shut_down()
+	//First alert condition: Overheat
+	if(temperature >= RBMK_TEMPERATURE_CRITICAL)
+		alert = TRUE
+		if(temperature >= RBMK_TEMPERATURE_MELTDOWN)
+			vessel_integrity -= (temperature / 100)
+			if(vessel_integrity <= temperature/100) //It wouldn't be able to tank another hit.
+				meltdown() //Oops! All meltdown
+				return
+	else
+		alert = FALSE
+	if(temperature < -200) //That's as cold as I'm letting you get it, engineering.
+		color = COLOR_CYAN
+		temperature = -200
+	else
+		color = null
+	//Second alert condition: Overpressurized (the more lethal one)
+	if(pressure >= RBMK_PRESSURE_CRITICAL)
+		alert = TRUE
+		shake_animation(0.5)
+		playsound(loc, 'sound/machines/clockcult/steam_whoosh.ogg', 100, TRUE)
+		var/turf/T = get_turf(src)
+		T.atmos_spawn_air("water_vapor=[pressure/100];TEMP=[CELSIUS_TO_KELVIN(temperature)]")
+		vessel_integrity -= (pressure/100)
+		if(vessel_integrity <= pressure/100) //It wouldn't be able to tank another hit.
+			blowout()
+			return
+	var/obj/structure/overmap/OM = get_overmap()
+	if(!OM) //Can't be bothered to do this any other way ;)
+		return
+	if(warning)
+		if(!alert) //Congrats! You stopped the meltdown / blowout.
+			OM?.stop_relay(CHANNEL_REACTOR_ALERT)
+			warning = FALSE
+			set_light(0)
+			light_color = LIGHT_COLOR_CYAN
+			set_light(10)
+	else
+		if(!alert)
+			return
+		if(world.time < next_warning)
```

Comment:
can we pull all the warning stuff into one var and setup a timer to repeat every thirty seconds if still required?

If done like that you can then have it all managed in one simple proc.

Reply:
Timers have far more performance overhead and are much more likely to break